source/github-events/on-react-page-update.tsx

```diff
@@ -0,0 +1,15 @@
+import {signalFromEvent} from 'abort-utils';
+
+export default function onReactPageUpdate(
+	callback: (signal: AbortSignal) => void,
+	signal: AbortSignal,
+): void {
+	document.addEventListener('soft-nav:payload', () => {
```

Comment:
This event appears on both clicks and history navigation, just on React pages.

In the future we might use a more standard https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API, but currently no Firefox nor Safari support.



### Click

<img width="811" alt="Screenshot 8" src="https://github.com/refined-github/refined-github/assets/1402241/e4603283-9a99-44d7-aea7-f4e1711be7da">


### History

<img width="489" alt="Screenshot 9" src="https://github.com/refined-github/refined-github/assets/1402241/5b4171a6-22d3-464d-900b-b0d4666b2c30">